docs/developer/decision-records/2024-10-09-vault-authentication-refactor/README.md

```diff
@@ -0,0 +1,72 @@
+# Decision Record Refactor of the HashiCorp Vault extension
+
+## Decision
+
+Refactor the HashiCorp Vault extension and make it extensible with different methods of authentication.
+
+## Rationale
+
+The current implementation of the authentication in the HashiCorp Vault extension has no way to add new authentication methods and to switch between them.
+It goes against the EDC extensibility model and needs to be remedied.
+Additionally, extracting the current implementation of the token authentication into its own extension will improve readability and maintainability of the code.
+
+## Affected Areas
+
+HashiCorp Vault extension
+
+## Approach
+
+To allow an extensible authentication module for HashiCorp Vault, an implementation similar to the existing management API authentication is possible.
+
+### Vault Authentication Service Interface
+
+To interact with a multitude of different authentication methods, an interface for the Vault Authentication Service is needed.
+This interface will contain two methods:
+
+* `login()`
+* `getVaultToken()`
+
+`login()` contains the authentication and fetches/generates the `client_token` from HashiCorp Vault.
+
+`getVaultToken()` returns the saved `client_token` when needed.S
+
+### Vault Authentication Service Extension
+
+The Vault Authentication Service Extensions each implement a single HashiCorp Vault Authentication method.
+To achieve this, each extension contains an implementation of the newly introduced Vault Authentication Service interface.
+The goal of each extension is to authenticate with HashiCorp Vault and to provide a valid `client_token` to the `HashicorpVaultClient`.
```

Comment:
Likely this will require to have an `spi` module to contain the interface, this should be documented here.